src/Cli/dotnet/CommonOptions.cs

```diff
@@ -93,13 +98,53 @@ public static Argument DefaultToCurrentDirectory(this Argument arg)
 
         public static Option DebugOption() => new Option<bool>("--debug");
 
+        public static Option SelfContainedOption() =>
+            new ForwardedOption<bool>(
+                "--self-contained",
+                CommonLocalizableStrings.SelfContainedOptionDescription)
+            .ForwardAsSingle(o => $"-property:SelfContained={o}");
+
+        public static Option NoSelfContainedOption() =>
```

Comment:
Do we really want to support both `--no-self-contained` and `--self-contained false`?

@KathleenDollard